examples/favorites_advanced/bloc_favorites_advanced/lib/feature_puppy/search/views/search_page.dart

```diff
@@ -0,0 +1,125 @@
+          child: ListView.builder(
+              padding: const EdgeInsets.only(bottom: 67),
+              itemCount: state.searchedPuppies!.length,
+              itemBuilder: (context, index) {
+                final item = state.searchedPuppies![index];
+                print('PuppyListStatus  : ${state.status}');
+                return PuppyCard(
+                  key: Key('${Keys.puppyCardNamePrefix}${item.id}'),
+                  onVisible: (puppy) => context
+                      .read<PuppyListBloc>()
+                      .add(PuppyFetchExtraDetailsEvent(puppy: puppy)),
+                  puppy: item,
+                  onCardPressed: (puppy) {},
+                  onFavorite: (puppy, isFavorite) {},
+                );
+              }),
+        ),
+      );
+
+  // RefreshIndicator puppyListStatusSuccess(
+  //         BuildContext context, PuppyListState state) =>
+  //     RefreshIndicator(
+  //       onRefresh: () {
+  //         context.read<PuppyListBloc>().add(ReloadPuppiesEvent());
+  //         return Future.delayed(const Duration(seconds: 1));
+  //       },
+  //       child: SafeArea(
+  //         child: ListView.builder(
+  //           padding: const EdgeInsets.only(bottom: 67),
+  //           itemCount: state.searchedPuppies!.length,
+  //           itemBuilder: (context, index) {
+  //             final item = state.searchedPuppies![index];
+  //             print('PuppyListStatus.success');
+  //
+  //             return PuppyCard(
+  //               key: Key('${Keys.puppyCardNamePrefix}${item.id}'),
+  //               puppy: item,
+  //               onCardPressed: (puppy) {},
+  //               onFavorite: (puppy, isFavorite) {},
+  //             );
+  //           },
+  //         ),
+  //       ),
+  //     );
+
+  // RefreshIndicator puppyListAllFetched(
+  //     BuildContext context, PuppyListState state) => RefreshIndicator(
+  //     onRefresh: () {
+  //       context.read<PuppyListBloc>().add(ReloadPuppiesEvent());
+  //       return Future.delayed(const Duration(seconds: 1));
+  //     },
+  //     child: SafeArea(
+  //       child: ListView.builder(
+  //         padding: const EdgeInsets.only(bottom: 67),
+  //         itemCount: state.searchedPuppies!.length,
+  //         itemBuilder: (context, index) {
+  //           final item = state.searchedPuppies![index];
+  //
+  //           return PuppyCard(
+  //             key: Key('${Keys.puppyCardNamePrefix}${item.id}'),
+  //             // onVisible: null,
+  //             puppy: item,
+  //             onCardPressed: (puppy) {},
+  //             onFavorite: (puppy, isFavorite) {},
+  //           );
+  //         },
+  //       ),
+  //     ),
+  //   );
+
+  Center puppyListStatusFailure() =>
+      const Center(child: Text('failed to fetch puppies'));
+
+// create error retry widget
+
+  LoadingWidget puppyListStatusInitial(BuildContext context) {
+    context.read<PuppyListBloc>().add(LoadPuppyListEvent());
```

Comment:
@StanislavPrime, dispatching events to the bloc from a build method it's not good practice. You may end up with too many dispatched `LoadPuppyListEvent` since the build method would be called very frequently.